packages/desktop-client/src/components/filters/FiltersMenu.js

```diff
@@ -205,62 +211,96 @@ function ConfigureField({
           {field === 'saved' && 'Existing filters will be cleared'}
         </View>
 
-        <Stack
-          direction="row"
-          align="flex-start"
-          spacing={1}
-          style={{ flexWrap: 'wrap' }}
-        >
-          {type === 'boolean'
-            ? [
-                <OpButton
-                  key="true"
-                  op="true"
-                  selected={value === true}
-                  onClick={() => {
-                    dispatch({ type: 'set-op', op: 'is' });
-                    dispatch({ type: 'set-value', value: true });
-                  }}
-                />,
-                <OpButton
-                  key="false"
-                  op="false"
-                  selected={value === false}
-                  onClick={() => {
-                    dispatch({ type: 'set-op', op: 'is' });
-                    dispatch({ type: 'set-value', value: false });
-                  }}
-                />,
-              ]
-            : ops.map(currOp => (
+        {type === 'boolean' ? (
+          <Stack
+            direction="row"
+            align="flex-start"
+            spacing={1}
+            style={{ flexWrap: 'wrap' }}
+          >
+            <OpButton
+              key="true"
+              op="true"
+              selected={value === true}
+              onClick={() => {
+                dispatch({ type: 'set-op', op: 'is' });
+                dispatch({ type: 'set-value', value: true });
+              }}
+            />
+            <OpButton
+              key="false"
+              op="false"
+              selected={value === false}
+              onClick={() => {
+                dispatch({ type: 'set-op', op: 'is' });
+                dispatch({ type: 'set-value', value: false });
+              }}
+            />
+          </Stack>
+        ) : (
+          [
+            <Stack
+              direction="row"
+              align="flex-start"
+              spacing={1}
+              style={{ flexWrap: 'wrap' }}
+            >
+              {ops.slice(0, 3).map(currOp => (
                 <OpButton
                   key={currOp}
                   op={currOp}
                   selected={currOp === op}
                   onClick={() => dispatch({ type: 'set-op', op: currOp })}
                 />
               ))}
-        </Stack>
+            </Stack>,
+            <Stack
+              direction="row"
+              align="flex-start"
+              spacing={1}
+              style={{ flexWrap: 'wrap' }}
+            >
+              {ops.slice(3, 6).map(currOp => (
```

Comment:
What happens if there are more than 6 items? Unfortunately we don’t have a prebuilt util for this but dynamically grouping the desired ops up into chunks of up to 3, and then dynamically generating row stacks. Alternatively, why didn’t using just one stack work?

Reply:
I've added an ops.length to capture anything higher than 6.

The goal was to have all the positive filters (is, contains, one of) on one row and the negative ones (is not, does not contain, not one of) on a separate row. Without this it looked weired to have "is not" up at the end of the first row and the other 2 on the second row. It's purely a look/feel call. Happy to remove it if needed.

Reply:
That sounds like a great change! I just didn’t understand why the code was that way but your explanation really helped!